ntbs-integration-tests/Alerts/DismissAlertTests.cs

```diff
@@ -0,0 +1,79 @@
+        {
+            // Arrange
+            var alertId = Utilities.ALERT_TO_DISMISS_ID;
+            var notificationOverviewPath = NotificationOverview(Utilities.NOTIFIED_ID);
+            var initialPage = await Client.GetAsync(notificationOverviewPath);
+            var initialDocument = await GetDocumentAsync(initialPage);
+            var alertElement = initialDocument.GetElementById($"alert-{alertId}");
+            Assert.NotNull(alertElement);
+
+            // Act
+            var response = await Client.SendVerificationPostAsync(initialPage, initialDocument, Dismiss(alertId), null);
+
+            // Assert
+            response.AssertRedirectTo(notificationOverviewPath);
+            var overviewPageAfterDismissal = await Client.GetAsync(notificationOverviewPath);
+            var documentAfterDismissal = await GetDocumentAsync(overviewPageAfterDismissal);
+            var alertElementAfterDismissal = documentAfterDismissal.GetElementById($"alert-{alertId}");
+            Assert.Null(alertElementAfterDismissal);
+        }
+
+        [Fact]
+        public async Task DismissClosedAlert_RedirectsToNotificationOverview()
+        {
+            // Arrange
+            var alertId = Utilities.CLOSED_ALERT_ID;
+            var notificationOverviewPath = NotificationOverview(Utilities.NOTIFIED_ID);
+            var initialPage = await Client.GetAsync(notificationOverviewPath);
+            var initialDocument = await GetDocumentAsync(initialPage);
+            var alertElement = initialDocument.GetElementById($"alert-{alertId}");
+            Assert.Null(alertElement);
+
+            // Act
+            var response = await Client.SendVerificationPostAsync(initialPage, initialDocument, Dismiss(alertId), null);
+
+            // Assert
+            response.AssertRedirectTo(notificationOverviewPath);
+            var overviewPageAfterDismissal = await Client.GetAsync(notificationOverviewPath);
+            var documentAfterDismissal = await GetDocumentAsync(overviewPageAfterDismissal);
+            var alertElementAfterDismissal = documentAfterDismissal.GetElementById($"alert-{alertId}");
+            Assert.Null(alertElementAfterDismissal);
+        }
+
+        [Fact]
+        public async Task AttemptToDismissNonExistentAlert_RedirectsToIndexPage()
+        {
+            // Arrange
+            var notAnAlertId = 999999;
+            var notificationOverviewPath = NotificationOverview(Utilities.NOTIFIED_ID);
+            var initialPage = await Client.GetAsync(notificationOverviewPath);
+            var initialDocument = await GetDocumentAsync(initialPage);
+            var alertElement = initialDocument.GetElementById($"alert-{notAnAlertId}");
+            Assert.Null(alertElement);
+
+            // Act
+            var response = await Client.SendVerificationPostAsync(initialPage, initialDocument, Dismiss(notAnAlertId), null);
+
+            // Assert
+            response.AssertRedirectTo("/");
```

Comment:
Thinking about it, it would probably be better if we returned a Bad Request in this case

Reply:
What would happen to the user in that case? If they don't see anything, then we could send bad request just so we can spot if this happens (and can look into why)

Reply:
I don't this case should ever happen, unless the user is doing something fishy (this is the case where a user gives an alert ID that doesn't exist in the database). So I don't really care what the user sees!